pkg/apis/admissionregistration/validation/validation.go

```diff
@@ -778,6 +784,47 @@ func validateNamedRuleWithOperations(n *admissionregistration.NamedRuleWithOpera
 	return allErrors
 }
 
+func validateMatchConditions(m []admissionregistration.MatchCondition, fldPath *field.Path) field.ErrorList {
+	var allErrors field.ErrorList
+	conditionNames := sets.NewString()
```

Comment:
limit the number here ... @jpbetz, opinion on a starting point? lower is better, easier to relax later, though we want to settle this before beta. be consistent with the limit for CEL admission conditions as well

Reply:
I had suggested 64 [here](https://github.com/kubernetes/kubernetes/pull/116261#discussion_r1132572844). But if we want to start low relax that probably to high of a start. The lowest I can imagine is around 20-30. If you go lower than that matchCondition authors can just get around it with "<condition1> && <condition2>", which doesn't benefit anyone.

Maybe 32? Thoughts?  We'll use the same value selected here in admission policy matchConditions.

Reply:
I set it at 64 for now but its easy to update once we have the number we want

Reply:
64 seems reasonable. It's really the total runtime that we care about, although I assume there's some compilation overhead for each individual expression.